### PR TITLE
New workflows for NWN

### PR DESCRIPTION
**Pull request by jimrojerTNO**
This adds two new workflows for the partners in NWN:

- The no heat losses end scenario sizing workflow, for the partners to iterate faster over results.
- The discounted capex workflow, allowing for taking into accounted interest on loans and inflation. For now we also do this one with no heat losses.

Workflows can be called with:
solution = run_end_scenario_sizing_no_heat_losses(EndScenarioSizingStagedHIGHS)
solution = run_end_scenario_sizing_no_heat_losses(EndScenarioSizingDiscountedHIGS)

NOTE!!!:
- main thing is to get the workflows in, we can later debug them fully.

### CODE REVIEW

File: src/rtctools_heat_network/workflows/grow_workflow.py [hunk 1]
```diff
@@ -73,6 +74,334 @@ def function(self, optimization_problem, ensemble_member):
...
+    ScenarioOutput,
+    TechnoEconomicMixin,
+    LinearizedOrderGoalProgrammingMixin,
+    SinglePassGoalProgrammingMixin,
+    ESDLMixin,
+    CollocatedIntegratedOptimizationProblem,
+):
+    """
+    Goal priorities are:
+    1. minimize TCO = Capex + Opex*lifetime
```
Comment (FJanssen-TNO) [inline src/rtctools_heat_network/workflows/grow_workflow.py:87]:
```suggestion
    1. Match heat demand with target
    2. minimize TCO = Capex + Opex*lifetime
```

File: src/rtctools_heat_network/workflows/grow_workflow.py [hunk 2]
```diff
@@ -73,6 +74,334 @@ def function(self, optimization_problem, ensemble_member):
         return optimization_problem.state(self.state)
 
 
+class EndScenarioSizingDiscountedHIGS(
```
Comment (FJanssen-TNO) [inline src/rtctools_heat_network/workflows/grow_workflow.py:77]: I would prefer to keep the same structure as we have for other problem definitions.
A main class which inherits all the Mixins and the problem and then a separate class to change the solversettings to HIGHS, which inherits from the main class

File: src/rtctools_heat_network/workflows/grow_workflow.py [hunk 3]
```diff
@@ -511,6 +840,43 @@ def solver_options(self):
...
+):
+    """
+    This function is used to run end_scenario_sizing problem. There are a few variations of the
+    same basic class. The main functionality this function adds is the staged approach, where
+    we first solve without heat_losses, to then solve the same problem with heat losses but
+    constraining the problem to only allow for the earlier found pipe classes and one size up.
+
+    This staged approach is done to speed up the problem, as the problem without heat losses is
+    much faster as it avoids inequality big_m constraints for the heat to discharge on pipes. The
+    one size up possibility is to avoid infeasibilities in compensating for the heat losses.
```
Comment (FJanssen-TNO) [inline src/rtctools_heat_network/workflows/grow_workflow.py:855]: update it: this is the problem without the heatlosses and no other stages

**Review by FJanssen-TNO — COMMENTED** [review]
I haven't gone trhough every detail (not checked if the test entails everything), but just commented on matters to ensure we keep a similar structure, workflow itself seems properly setup.

File: src/rtctools_heat_network/workflows/grow_workflow.py (hunk 3 above)
Comment (jimrojerTNO) [reply src/rtctools_heat_network/workflows/grow_workflow.py:856]: done

File: src/rtctools_heat_network/workflows/grow_workflow.py (hunk 2 above)
Comment (jimrojerTNO) [reply src/rtctools_heat_network/workflows/grow_workflow.py]: done

**Comment by jimrojerTNO** [comment]
@FJanssen-TNO can you see if it ok now and press merge if it is?

**Review by FJanssen-TNO — APPROVED** [review]

**Pull request merged**